resources/metadata/whitesource.yaml

```diff
@@ -0,0 +1,523 @@
+metadata:
+    name: whitesourceExecuteScan
+    description: BETA
+    longDescription: |-
+      BETA
+      With this step [WhiteSource](https://www.whitesourcesoftware.com) security and license compliance scans can be executed and assessed.
+      WhiteSource is a Software as a Service offering based on a so called unified agent that locally determines the dependency
+      tree of a node.js, Java, Python, Ruby, or Scala based solution and sends it to the WhiteSource server for a policy based license compliance
+      check and additional Free and Open Source Software Publicly Known Vulnerabilities detection.
+      !!! note "Docker Images"
+          The underlying Docker images are public and specific to the solution's programming language(s) and therefore may have to be exchanged
+          to fit to and support the relevant scenario. The default Python environment used is i.e. Python 3 based.
+      !!! warn "Restrictions"
+          Currently the step does contain hardened scan configurations for `scanType` `'pip'` and `'go'`. Other environments are still being elaborated,
+          so please thoroughly check your results and do not take them for granted by default.
+          Also not all environments have been thoroughly tested already therefore you might need to tweak around with the default containers used or
+          create your own ones to adequately support your scenario. To do so please modify `dockerImage` and `dockerWorkspace` parameters.
+          The step expects an environment containing the programming language related compiler/interpreter as well as the related build tool. For a list
+          of the supported build tools per environment please refer to the [WhiteSource Unified Agent Documentation](https://whitesource.atlassian.net/wiki/spaces/WD/pages/33718339/Unified+Agent).
+spec:
+  inputs:
+    params:
+    - name: buildDescriptorFile
+      type: string
+      description: Explicit path to the build descriptor file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./Gopkg.toml
+      conditions:
+      - conditionRef: strings-equal
+        params:
+        - name: scanType
```

Comment:
@OliverNocon is this supported? It increases the `yaml` size dramatically and makes it harder to read.

Reply:
I think this is okay.
@nevskrem ?